accepted/1847 - FinalizationRegistry/proposal.md

```diff
@@ -247,15 +238,27 @@ abstract class Finalizable {
 typedef NativeFinalizer = Void Function(Pointer<Void>);
 typedef NativeFinalizerPtr = Pointer<NativeFunction<NativeFinalizer>>
 
-/// [FinalizationRegistry] which will execute its finalizers as early as
-/// possible without waiting for control to return to the event loop.
+/// [Finalizer] *may* call the given callback when objects it is attached to
```

Comment:
Not a good first line. Describe the instances of this class, using noun phrases.
This line is describing a completely different class.
In some cases (doc overview sections), this is the *only* line you see describing the class.

> /// A native finalizer which can be attached to Dart objects.

Then describe when and how to use it:

> ///
> /// When [attach]ed to a Dart object, this finalizer's native callback is called
> /// after the Dart object is garbage collected or becomes inaccessible for other
> /// reasons.
> ///
> /// Callbacks will happen as early as possible, when the object becomes 
> /// inaccessible to the program,
> /// and may happen at any moment during execution of the program.
> /// At the latest, when an isolate group shuts down, 
> /// this callback is guaranteed to be called for each object in that isolate group
> /// that the finalizer is still attached to.
> ///
> /// If a callback is no longer desired, the [detach] method can remove 
> /// any number of attachments to objects by providing the detachment key 
> /// used when the finalizer was attached to those objects.

Then add perspective:

> /// 
> /// Compared to the [Finalizer] from `dart:core`,
> /// which makes no promises to ever call an attached `callback`,
> /// this native finalizer promises that all attached finalizers are 
> /// definitely called at least once before the program ends,
> /// and the callbacks are called as soon as possible after an object 
> /// is recognized as inaccessible.

(The reader should be able to stop at any time and have gotten the most important parts of the documentation.)

Reply:
Applied. 

Though I find it a bit redundant to document class called `NativeFinalizer` as `A native finalizer which can be attached to Dart objects.`. It's all in the name already.